neps/utils/cli.py

```diff
@@ -52,35 +55,62 @@
 warnings.filterwarnings("ignore", category=UserWarning, module="torch._utils")
 
 
-def get_root_directory(args: argparse.Namespace) -> Path | None:
-    """Load the root directory from the provided argument or from the config.yaml file."""
+def validate_directory(path: Path) -> bool:
+    """
+    Validates whether the given path exists and is a directory.
+
+    Args:
+        path (Path): The path to validate.
+
+    Returns:
+        bool: True if valid, False otherwise.
+    """
+    if not path.exists():
+        print(f"Error: The directory '{path}' does not exist.")
+        return False
+    if not path.is_dir():
+        print(f"Error: The path '{path}' exists but is not a directory.")
+        return False
+    return True
+
+
+def get_root_directory(args: argparse.Namespace) -> Optional[Path]:
+    # Command-line argument handling
     if args.root_directory:
-        return Path(args.root_directory)
+        root_dir = Path(args.root_directory)
+        if validate_directory(root_dir):
+            return root_dir
+        else:
+            return None
 
+    # Configuration file handling
     config_path = Path("run_config.yaml")
     if config_path.exists():
-        with config_path.open("r") as file:
-            config = yaml.safe_load(file)
+        try:
+            with config_path.open("r") as file:
+                config = yaml.safe_load(file)
+        except yaml.YAMLError as e:
+            print(f"Error parsing '{config_path}': {e}")
+            return None
+
         root_directory = config.get(ROOT_DIRECTORY)
         if root_directory:
             root_directory_path = Path(root_directory)
-            if not root_directory_path.exists() or not root_directory_path.is_dir():
-                print(
-                    f"Error: The directory {root_directory_path} does not exist or is "
-                    f"not a "
-                    f"directory."
-                )
+            if validate_directory(root_directory_path):
+                return root_directory_path
+            else:
                 return None
-            return root_directory_path
         else:
-            raise ValueError(
-                "The config.yaml file exists but does not contain 'root_directory'."
+            print(
+                "Error: The 'run_config.yaml' file exists but does not contain the 'root_directory' key."
             )
+            return None
     else:
-        raise ValueError(
-            "Either the root_directory must be provided as an argument or config.yaml "
-            "must exist with a 'root_directory' key."
+        print(
+            "Error: 'root_directory' must be provided as a command-line argument "
+            "or defined in 'run_config.yaml'."
         )
+        return None
```

Comment:
One thing that might make this a bit more informative is to call `config_path = Path("run_config.yaml").resolve()` and to use this path in the errors printed. Gives a bit more information on the complete path that is being looked for.